app/utils/nonsipp/ValidationUtils.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils.nonsipp
+
+import pages.nonsipp.schemedesignatory.{ActiveBankAccountPage, HowManyMembersPage, WhyNoBankAccountPage}
+import config.RefinedTypes.{Max50, Max5000}
+import models.SchemeId.Srn
+import pages.nonsipp.accountingperiod.AccountingPeriods
+import pages.nonsipp.CheckReturnDatesPage
+import models.UserAnswers
+
+object ValidationUtils {
+
+  /**
+   * This method determines whether or not a return is missing any necessary answers by calling the validation
+   * method for each section in turn.
+   * @param userAnswers The answers provided by the user which are checked for validity
+   * @param srn The Scheme Reference Number, used for the .get calls in the section validation methods
+   * @return true if all answers in all sections are valid, false if any necessary answers in any section are missing
+   */
+  def validateAllSections(userAnswers: UserAnswers, srn: Srn): Boolean =
+    validateBasicDetailsSection(userAnswers, srn) &&
+      //...
+      validateLandOrPropertySection(userAnswers, srn)
+  //...
+
+  /**
+   * This method determines whether or not the Basic Details section is missing any necessary answers.
+   * @param userAnswers The answers provided by the user which are checked for validity
+   * @param srn The Scheme Reference Number, used for the .get calls
+   * @return true if all answers in this section are valid, false if any necessary answers are missing
+   */
+  def validateBasicDetailsSection(userAnswers: UserAnswers, srn: Srn): Boolean = {
```

Comment:
It used to be the case that the user could wipe the accounting periods by going back and changing their answers, though admittedly it was a bit of an edge case, and I don't recall whether or not it POSTed successfully and/or persisted. I just tried it locally and it errored out instead, so in any case it seems like that's not possible now.

Initially I thought this method would be necessary as part of the validation carried out on the declaration & submission page, but that'll depend on how we decide to implement it, so yes - it might be that this method isn't used in the long run, but it made for a good starting block.